File: backend/src/main/java/ca/metricalsky/winston/entity/view/CommentCountView.java
package ca.metricalsky.winston.entity.view;

public interface CommentCountView {

    String getVideoId();

    default Long getComments() {
        return getCommentsAndReplies() - getReplies();
    }

    Long getCommentsAndReplies();

    Long getReplies();

    Long getTotalReplies();

    class Empty implements CommentCountView {

        @Override
        public String getVideoId() {
            return "";
        }

        @Override
        public Long getCommentsAndReplies() {
            return 0L;
        }

        @Override
        public Long getReplies() {
            return 0L;
        }

        @Override
        public Long getTotalReplies() {
            return 0L;
        }
    }
}

File: database/src/main/kotlin/ca/metricalsky/winston/database/view/CommentCountView.kt
package ca.metricalsky.winston.database.view

interface CommentCountView {

    val videoId: String
    val comments: Long
        get() = this.commentsAndReplies - this.replies
    val commentsAndReplies: Long
    val replies: Long
    val totalReplies: Long
}
